plugins/parsers/prometheusremotewrite/parser.go

```diff
@@ -0,0 +1,101 @@
+package prometheusremotewrite
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/gogo/protobuf/proto"
+	"github.com/prometheus/common/model"
+)
+
+type Parser struct {
+	DefaultTags map[string]string
+}
+
+func (p *Parser) Parse(buf []byte) ([]telegraf.Metric, error) {
+	var metrics []telegraf.Metric
+	var err error
+	var req prompb.WriteRequest
+
+
+	if err := proto.Unmarshal(buf, &req); err != nil {
+		return nil, fmt.Errorf("unable to unmarshal request body: %s", err)
+	}
+
+
+	now := time.Now()
+
+	for _, ts := range req.Timeseries {
+		tags := map[string]string{}
+		for key, value := range p.DefaultTags {
+			tags[key] = value
+		}
+
+		for _, l := range ts.Labels {
+			tags[l.Name] = l.Value
+		}
+
+		metricName := tags[model.MetricNameLabel]
+		delete(tags, model.MetricNameLabel)
+
+		for _, s := range ts.Samples {
+			fields := getNameAndValue(&s, metricName)
+
+			// converting to telegraf metric
+			if len(fields) > 0 {
+				t := getTimestamp(&s, now)
+				metric, err := metric.New("prometheusremotewrite", tags, fields, t)
+				if err == nil {
```

Comment:
We might silently drop metrics here without allowing the user to be aware of this. Please either fail here (i.e. `return err`) or _at least_ issue a log message in case we see an error here.